terraform/import/import.go

```diff
@@ -217,36 +238,43 @@ func convertToHCLLine(input interface{}, indentLevel int, builder *strings.Build
 	}
 	var m map[string]interface{}
 	json.Unmarshal(b, &m)
-
 	for k, v := range m {
-		switch reflect.TypeOf(v).Kind() {
-		case reflect.String:
-			builder.WriteString(fmt.Sprintf("%s%s = %q\n", indent(indentLevel), utils.ToSnakeCase(k), v))
-		case reflect.Int, reflect.Int32, reflect.Float32, reflect.Float64, reflect.Bool:
-			builder.WriteString(fmt.Sprintf("%s%s = %v\n", indent(indentLevel), utils.ToSnakeCase(k), v))
-		case reflect.Array, reflect.Slice:
-			sl := v.([]interface{})
-			if len(sl) > 0 {
-				switch reflect.TypeOf(sl[0]).Kind() {
-				case reflect.Array, reflect.Slice, reflect.Map:
-					for j := 0; j < len(sl); j++ {
-						builder.WriteString(strings.ToLower(fmt.Sprintf("\n%s%s {\n", indent(indentLevel), utils.ToSnakeCase(k))))
-						convertToHCLLine(sl[j], indentLevel+1, builder)
-						builder.WriteString(fmt.Sprintf("%s}\n", indent(indentLevel)))
-					}
-				default:
-					builder.WriteString(fmt.Sprintf("%s%s = [", indent(indentLevel), utils.ToSnakeCase(k)))
-					for j := 0; j < len(sl); j++ {
-						builder.WriteString(fmt.Sprintf("%q", sl[j]))
-						if j < len(sl)-1 {
-							builder.WriteString(",")
+		if v != nil {
+			switch reflect.TypeOf(v).Kind() {
+			case reflect.String:
+				builder.WriteString(fmt.Sprintf("%s%s = %q\n", indent(indentLevel), utils.ToSnakeCase(k), v))
+			case reflect.Int, reflect.Int32, reflect.Float32, reflect.Float64, reflect.Bool:
+				builder.WriteString(fmt.Sprintf("%s%s = %v\n", indent(indentLevel), utils.ToSnakeCase(k), v))
+			case reflect.Array, reflect.Slice:
+				sl := v.([]interface{})
+				if len(sl) > 0 {
+					switch reflect.TypeOf(sl[0]).Kind() { // array of complex stuff
+					case reflect.Array, reflect.Slice, reflect.Map:
+						for j := 0; j < len(sl); j++ {
+							builder.WriteString(strings.ToLower(fmt.Sprintf("\n%s%s {\n", indent(indentLevel), utils.ToSnakeCase(k))))
+							convertToHCLLine(sl[j], indentLevel+1, builder)
+							builder.WriteString(fmt.Sprintf("%s}\n", indent(indentLevel)))
 						}
+					default: // array of strings
+						builder.WriteString(fmt.Sprintf("%s%s = [", indent(indentLevel), utils.ToSnakeCase(k)))
+						for j := 0; j < len(sl); j++ {
+							builder.WriteString(fmt.Sprintf("%q", sl[j]))
+							if j < len(sl)-1 {
+								builder.WriteString(",")
+							}
+						}
+						builder.WriteString("]\n")
 					}
-					builder.WriteString("]\n")
 				}
```

Comment:
Nested switch makes me uneasy. Can this be broken up?

Reply:
I'll think about it when I refactor this part. This had to be inserted because complex types containing complex types like lists of lists and lists of objects are accepted as well as  complex types containing primitives